SofaKernel/modules/SofaBaseTopology/src/SofaBaseTopology/TopologyData.h

```diff
@@ -135,6 +116,18 @@ class TopologyData : public sofa::core::topology::BaseTopologyData<VecT>
     virtual void removeOnMovedPosition(const sofa::helper::vector<Index>& indices);
 
 
+
+    ////////////////////////////////////// DEPRECATED ///////////////////////////////////////////
+    SOFA_ATTRIBUTE_DISABLED("v21.06", "PR#2082", "This method has been removed as it is not part of the new topology change design.")
+    void addInputData(sofa::core::objectmodel::BaseData* _data) = delete;
+
+    SOFA_ATTRIBUTE_DISABLED("v21.06", "PR#2082", "This method was deleted because it presented risks. Use Write/Read Accessor instead.")
+    const value_type& operator[](int i) const = delete;
+
+    SOFA_ATTRIBUTE_DISABLED("v21.06", "PR#2082", "This method was deleted because it presented risks. Use Write/Read Accessor instead.")
+    value_type& operator[](int i) = delete;
```

Comment:
```suggestion
    SOFA_ATTRIBUTE_DISABLED("v21.06 (PR#2082)", "v21.06 (PR#2082)", "This method has been removed as it is not part of the new topology change design.")
    void addInputData(sofa::core::objectmodel::BaseData* _data) = delete;

    SOFA_ATTRIBUTE_DISABLED("v21.06 (PR#2082)", "v21.06 (PR#2082)", "This method was deleted because it presented risks. Use Write/Read Accessor instead.")
    const value_type& operator[](int i) const = delete;

    SOFA_ATTRIBUTE_DISABLED("v21.06 (PR#2082)", "v21.06 (PR#2082)", "This method was deleted because it presented risks. Use Write/Read Accessor instead.")
    value_type& operator[](int i) = delete;
```